packages/kinvey-datastore/src/repositories/offline-repositories/inmemory-offline-repository.js

```diff
@@ -0,0 +1,203 @@
+import keyBy from 'lodash/keyBy';
+import { NotFoundError } from 'kinvey-errors';
+
+import { OfflineRepository } from '../offline-repository';
+import {
+  applyQueryToDataset,
+  collectionsMaster as masterCollectionName
+} from '../utils';
+import { ensureArray } from '../../utils';
+
+// Imported for typings
+// import { KeyValuePersister } from '../../persisters';
+
+export class InmemoryOfflineRepository extends OfflineRepository {
+  /** @type {KeyValuePersister} */
+  _persister;
+  /** @type {PromiseQueue} */
+  _queue;
+
+  constructor(persister, promiseQueue) {
+    super();
+    this._persister = persister;
+    this._queue = promiseQueue;
+  }
+
+  // ----- private methods
+
+  // TODO: integrate with db/collection concept in parent(s)?
+  _formCollectionKey(collection) {
+    const appKey = this._getAppKey();
+    return `${appKey}${collection}`;
+  }
+
+  _readAll(collection) {
+    const key = this._formCollectionKey(collection);
+    return this._persister.readEntities(key)
+      .then(entities => entities || []);
+  }
+
+  // TODO: Keep them by id
+  _saveAll(collection, entities) {
+    const key = this._formCollectionKey(collection);
+    return this._persister.persistEntities(key, entities);
+  }
+
+  _deleteAll(collection) {
+    const key = this._formCollectionKey(collection);
+    return this._persister.deleteEntities(key);
+  }
+
+  _enqueueCrudOperation(collection, operation) {
+    const key = this._formCollectionKey(collection);
+    return this._queue.enqueue(key, operation);
+  }
+
+  _getAllCollections() {
+    return this._readAll(masterCollectionName);
+  }
+
+  _countAndDelete(collection) {
+    return this._readAll(collection)
+      .then((entities) => {
+        return this._deleteAll(collection)
+          .then(() => entities.length);
+      });
+  }
+
+  _clearAllCollections() {
+    return this._getAllCollections()
+      .then((collections) => {
+        const promises = collections.map(c => this._deleteAll(c));
+        return Promise.all(promises);
+      })
+      .then(() => this._deleteAll(masterCollectionName));
+  }
+
+  // ----- protected methods
+  _ensureCollection(collection) {
```

Comment:
This method name doesn't say much about what it is supposed to do. A comment or a better name would help.

Reply:
👍 
Will rename it to `_ensureCollectionExists()`.